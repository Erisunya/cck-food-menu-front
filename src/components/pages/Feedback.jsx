import styles from "./Feedback.module.css";
import { useForm } from "react-hook-form";

const Feedback = () => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  const onSubmit = (data) => {
    console.log(data);
  };

  return (
    <div className={styles.feedback}>
      <form onSubmit={handleSubmit(onSubmit)} className={styles.form}>
        <p className={styles.formTitle}>Leave your feedback here!</p>
        <div className={styles.inputContainer}>
          <label htmlFor="name">Name</label>
          <input
            placeholder="Joseph Tan Ah Kow"
            type="text"
            id="name"
            {...register("name", { required: true })}
          />
        </div>

        <div className={styles.inputContainer}>
          <label htmlFor="email">Email</label>
          <input
            placeholder="josepthtaw@gmail.com"
            type="text"
            id="email"
            {...register("email", { pattern: /^\S+@\S+$/i })}
          />
        </div>

        <div className={styles.inputContainer}>
          <label htmlFor="telegram">Telegram</label>
          <input
            placeholder="@josepthtaw"
            type="text"
            id="telegram"
            {...register("telegram")}
          />
        </div>

        <div className={styles.inputContainer}>
          <label htmlFor="feedback">Feedback</label>
          <textarea
            placeholder="Can you please update the menu from ABC Chicken Rice to show the latest price?"
            id="feedback"
            rows={4}
            {...register("feedback", { required: true })}
          />
        </div>

        <button type="submit" className={styles.submit}>
          Submit
        </button>
      </form>
    </div>
  );
};

export default Feedback;
