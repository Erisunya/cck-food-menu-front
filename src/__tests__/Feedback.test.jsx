import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Feedback from "../components/pages/Feedback";

describe("Feedback component", () => {
  it("renders feedback form correctly", () => {
    const component = render(<Feedback />);

    expect(component).toMatchSnapshot();
  });

  it("enables the submit button when form is filled out", async () => {
    render(<Feedback />);

    expect(document.querySelector("button").disabled).toBeTruthy();

    await userEvent.type(
      screen.getByRole("textbox", { name: "Name*" }),
      "Alex Lim"
    );

    expect(document.querySelector("button").disabled).toBeFalsy();
  });

  it("shows an error message when the required fields are not filled out", async () => {
    render(<Feedback />);

    await userEvent.type(
      screen.getByRole("textbox", { name: "Email" }),
      "test@email.com"
    );
    await userEvent.click(screen.getByRole("button"));

    expect(screen.getByText("Name is required.")).toBeVisible();
    expect(screen.getByText("Feedback is required.")).toBeVisible();
  });
});
