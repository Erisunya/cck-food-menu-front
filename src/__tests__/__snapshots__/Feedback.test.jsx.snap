// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Feedback component > renders feedback form correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_feedback_2acf0e"
      >
        <form
          class="_form_2acf0e"
        >
          <p
            class="_formTitle_2acf0e"
          >
            Leave your feedback here!
          </p>
          <div
            class="_inputContainer_2acf0e"
          >
            <label
              for="name"
            >
              Name
            </label>
            <input
              id="name"
              name="name"
              placeholder="Joseph Tan Chee Guan"
              type="text"
            />
          </div>
          <div
            class="_inputContainer_2acf0e"
          >
            <label
              for="feedback"
            >
              Feedback
            </label>
            <textarea
              id="feedback"
              name="feedback"
              placeholder="Can you please update the menu from ABC Chicken Rice to show the latest price?"
              rows="4"
            />
          </div>
          <p
            class="_contactText_2acf0e"
          >
            If you'd like to be contacted regarding your feedback, please leave your email and/or Telegram handle below as well.
          </p>
          <div
            class="_inputContainer_2acf0e"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              id="email"
              name="email"
              placeholder="josepthtcg@gmail.com"
              type="text"
            />
          </div>
          <div
            class="_inputContainer_2acf0e"
          >
            <label
              for="telegram"
            >
              Telegram
            </label>
            <input
              id="telegram"
              name="telegram"
              placeholder="@josepthtcg"
              type="text"
            />
          </div>
          <button
            class="_submit_2acf0e"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_feedback_2acf0e"
    >
      <form
        class="_form_2acf0e"
      >
        <p
          class="_formTitle_2acf0e"
        >
          Leave your feedback here!
        </p>
        <div
          class="_inputContainer_2acf0e"
        >
          <label
            for="name"
          >
            Name
          </label>
          <input
            id="name"
            name="name"
            placeholder="Joseph Tan Chee Guan"
            type="text"
          />
        </div>
        <div
          class="_inputContainer_2acf0e"
        >
          <label
            for="feedback"
          >
            Feedback
          </label>
          <textarea
            id="feedback"
            name="feedback"
            placeholder="Can you please update the menu from ABC Chicken Rice to show the latest price?"
            rows="4"
          />
        </div>
        <p
          class="_contactText_2acf0e"
        >
          If you'd like to be contacted regarding your feedback, please leave your email and/or Telegram handle below as well.
        </p>
        <div
          class="_inputContainer_2acf0e"
        >
          <label
            for="email"
          >
            Email
          </label>
          <input
            id="email"
            name="email"
            placeholder="josepthtcg@gmail.com"
            type="text"
          />
        </div>
        <div
          class="_inputContainer_2acf0e"
        >
          <label
            for="telegram"
          >
            Telegram
          </label>
          <input
            id="telegram"
            name="telegram"
            placeholder="@josepthtcg"
            type="text"
          />
        </div>
        <button
          class="_submit_2acf0e"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
